src/main/java/com/limechain/network/protocol/sync/SyncProtocol.java

```diff
@@ -21,25 +26,35 @@ public SyncProtocol() {
         super(MAX_REQUEST_SIZE, MAX_RESPONSE_SIZE);
     }
 
-    @Override
-    protected CompletableFuture<SyncController> onStartInitiator(Stream stream) {
+    @NotNull
```

Comment:
Do we need the `@NotNull`? It's only a cosmetic annotation if we aren't using some kind of Intellij static analysis.

Reply:
Goes for the `@NotNull`s in method args as well.

Reply:
Well, we don't really need it, but the static analyzer complains because the methods onStartInitiator and onStartResponder both override a method that has NotNull annotation. It's pretty much the same as skipping the Override annotation on one of both methods. It will work without them but you will be safer with them,